File: client/src/app/models/student.ts
import { Classroom } from './classroom';

export class Student{
    id: number;
    name: string;
    address: string;
    dob: Date;
    classroomID: number;
    classroom: Classroom;
    /**
     *
     */
    constructor() {
        this.id = 0;
        this.name = '';
        this.dob = new Date();
        this.address = '';
        this.classroomID = 0;
        
    }
}

File: server/server/Models/Student.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace server.Models
{
    public class Student : CRUDCommon
    {
        public string Name { get; set; }
        public string Address { get; set; }
        public long ClassroomID { get; set; }
        public DateTime DOB { get; set; }

        [ForeignKey("ClassroomID")]
        public virtual Classroom Classroom { get; set; }
    }
}
